magma/primitives/mux.py

```diff
@@ -185,15 +185,18 @@ def dict_lookup(dict_, select, default=0):
     return output
 
 
-def list_lookup(list_, select, default=0):
+def list_lookup(list_, select, default=None):
     """
     Use `select` as an index into `list` (similar to a case statement)
 
     `default` is used when `select` does not match any of the indices (e.g.
-    when the select width is longer than the list) and has a default value of
-    0.
+    when the select width is longer than the list).  If it is `None`, the last
+    element of the list will be used.
     """
     output = default
-    for i, elem in enumerate(list_):
-        output = mux([output, elem], i == select)
+    if default is None:
+        output = list_[-1]
+        list_ = list_[:-1]
+    for i in range(len(list_) - 1, -1, -1):
```

Comment:
Can we do

```python
for i, elt in reversed(enumerate(list_)):
```
?

Reply:
and also add an inline comment here like

```
# We chain the muxes in reverse order so that the emitted Verilog is in forward order.
```

Reply:
> ```python
> for i, elt in reversed(enumerate(list_)):
> ```

I tried this originally and enumerate objects are not reversible (presumably because reversing them would require expanding them entirely).

```
>>> for i, elt in reversed(enumerate(list_)):
...
KeyboardInterrupt
>>> list_ = [x for x in range(9)]
>>> for i, elt in reversed(enumerate(list_)):
...     print(i, elt)
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'enumerate' object is not reversible
```